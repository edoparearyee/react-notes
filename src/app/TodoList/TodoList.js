import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './TodoList.scss';
import Todo from '../Todo/Todo';

class TodoList extends Component {
  render() {
    return (
      <div>
        {this.props.items.map(item => (
          <Todo
            key={item.id}
            id={item.id}
            title={item.title}
            complete={item.complete}
            onChange={this.props.onChange}
          />
        ))}
      </div>
    );
  }
}

TodoList.propTypes = {
  items: PropTypes.array.isRequired,
  onChange: PropTypes.func,
};

export default TodoList;
