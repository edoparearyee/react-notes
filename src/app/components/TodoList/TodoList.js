import React, { Component } from 'react';
import PropTypes from 'prop-types';
import List from '@material-ui/core/List';

import './TodoList.scss';
import Todo from '../Todo/Todo';

class TodoList extends Component {
  render() {
    return (
      <List className="TodoList">
        {this.props.items.map((item, i) => (
          <Todo
            key={i}
            {...item}
            onChange={this.props.onChange}
            onToggle={this.props.onToggle}
            onDelete={this.props.onDelete}
          />
        ))}
      </List>
    );
  }
}

TodoList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      complete: PropTypes.bool,
    }),
  ).isRequired,
  onChange: PropTypes.func,
  onToggle: PropTypes.func,
  onDelete: PropTypes.func,
};

export default TodoList;
