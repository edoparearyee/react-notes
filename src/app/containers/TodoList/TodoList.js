import { connect } from 'react-redux';
import { todoEdit, todoToggle, todoDelete } from '../../actions/todos';

import TodoList from '../../components/TodoList/TodoList';

const mapStateToProps = state => {
  return {
    items: state.todos.items,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onChange: ({ id, title }) => {
      dispatch(todoEdit(id, title));
    },
    onToggle: ({ id }) => {
      dispatch(todoToggle(id));
    },
    onDelete: ({ id }) => {
      dispatch(todoDelete(id));
    },
  };
};

const TodoListContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(TodoList);

export default TodoListContainer;
