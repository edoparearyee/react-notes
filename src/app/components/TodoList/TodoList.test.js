import React from 'react';
import { shallow } from 'enzyme';

import TodoList from './TodoList';

const mockTodos = [
  {
    id: '6d3e6088-bc4b-a1b1-b0a2-61328db3ba6',
    title: 'Culpa excepteur nisi fugiat',
    complete: false,
  },
  {
    id: 'a4a14b66-b53b-8511-2ba8-7203570f5cd5',
    title: 'Et aliqua enim aliqua id',
    complete: false,
  },
];

describe('TodoList', () => {
  it('renders without crashing', () => {
    const component = shallow(<TodoList items={mockTodos} />);
    expect(component).toMatchSnapshot();
  });
});
