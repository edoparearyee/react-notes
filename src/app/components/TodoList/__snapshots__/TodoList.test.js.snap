// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders without crashing 1`] = `
<WithStyles(List)
  className="TodoList"
>
  <Todo
    complete={false}
    id="6d3e6088-bc4b-a1b1-b0a2-61328db3ba6"
    key="0"
    title="Culpa excepteur nisi fugiat"
  />
  <Todo
    complete={false}
    id="a4a14b66-b53b-8511-2ba8-7203570f5cd5"
    key="1"
    title="Et aliqua enim aliqua id"
  />
</WithStyles(List)>
`;
